[Bug]: make dev-env-setup not working as expected

### Prerequisites

- [x] I searched existing issues
- [x] I can reproduce this issue

### Bug Description

`make dev-env-setup` failing with `Error 8` without any error message

### Component

Deployment/Config

### Steps to Reproduce

```
$ make dev-env-setup
Setting up NVSentinel development environment...

╔════════════════════════════════════════════════════════╗
║   NVSentinel Development Environment Setup            ║
╚════════════════════════════════════════════════════════╝

ℹ️  Platform: linux-amd64

✅ yq already installed: yq (https://github.com/mikefarah/yq/) version v4.46.1
ℹ️  Loading versions from .versions.yaml...

ℹ️  Target Versions:
  Go:              1.25
  Python:          3.10
  Poetry:          1.8.2
  golangci-lint:   v2.5.0
  protobuf:        v33.0
  black:           25.9.0
  shellcheck:      v0.11.0

════════════════════════════════════════════════════════
ℹ️  Go Installation
════════════════════════════════════════════════════════
ℹ️  Current Go version: 1.25.3
✅ Go 1.25 already installed

════════════════════════════════════════════════════════
ℹ️  Docker Check
════════════════════════════════════════════════════════
✅ Docker is installed and running: 28.5.2

════════════════════════════════════════════════════════
ℹ️  Python & Poetry Installation
════════════════════════════════════════════════════════
✅ Python installed: 3.12
✅ Poetry installed: 1.8.2
✅ Black installed: 25.9.0
3.12.3

════════════════════════════════════════════════════════
ℹ️  Development Tools Installation
════════════════════════════════════════════════════════
✅ Helm already installed: v3.19.0+g3d8990f
✅ kubectl already installed: Client Version: v1.34.1
Kustomize Version: v5.7.1
ℹ️  Installing Protocol Buffers v33.0...
Continue? [Y/n] Y
make: *** [Makefile:123: dev-env-setup] Error 8
```

I get the same error message for these key combination:
1. `y`
2. `caps` + `y`
3. `shift` + `y`
4. `enter`

### Environment

- NVSentinel version: latest
- Distro: Ubuntu 24.04.3 LTS

> [@XRFXLP](https://github.com/XRFXLP) can you repro this with `make dev-env-setup AUTO_MODE=true`?

with `y`:
```
$ make dev-env-setup AUTO_MODE=true
Setting up NVSentinel development environment...

╔════════════════════════════════════════════════════════╗
║   NVSentinel Development Environment Setup            ║
╚════════════════════════════════════════════════════════╝

ℹ️  Platform: linux-amd64

✅ yq already installed: yq (https://github.com/mikefarah/yq/) version v4.46.1
ℹ️  Loading versions from .versions.yaml...

ℹ️  Target Versions:
  Go:              1.25
  Python:          3.10
  Poetry:          1.8.2
  golangci-lint:   v2.5.0
  protobuf:        v33.0
  black:           25.9.0
  shellcheck:      v0.11.0

════════════════════════════════════════════════════════
ℹ️  Go Installation
════════════════════════════════════════════════════════
ℹ️  Current Go version: 1.25.4
✅ Go 1.25 already installed

════════════════════════════════════════════════════════
ℹ️  Docker Check
════════════════════════════════════════════════════════
✅ Docker is installed and running: 29.0.2

════════════════════════════════════════════════════════
ℹ️  Python & Poetry Installation
════════════════════════════════════════════════════════
✅ Python installed: 3.12
✅ Poetry installed: 1.8.2
✅ Black installed: 25.9.0
3.12.3

════════════════════════════════════════════════════════
ℹ️  Development Tools Installation
════════════════════════════════════════════════════════
✅ Helm already installed: v3.19.2+g8766e71
✅ kubectl already installed: Client Version: v1.34.1
Kustomize Version: v5.7.1
ℹ️  Installing Protocol Buffers v33.0...
Continue? [Y/n] y
make: *** [Makefile:123: dev-env-setup] Error 8
```

With empty:
```
$ make dev-env-setup AUTO_MODE=true
Setting up NVSentinel development environment...

╔════════════════════════════════════════════════════════╗
║   NVSentinel Development Environment Setup            ║
╚════════════════════════════════════════════════════════╝

ℹ️  Platform: linux-amd64

✅ yq already installed: yq (https://github.com/mikefarah/yq/) version v4.46.1
ℹ️  Loading versions from .versions.yaml...

ℹ️  Target Versions:
  Go:              1.25
  Python:          3.10
  Poetry:          1.8.2
  golangci-lint:   v2.5.0
  protobuf:        v33.0
  black:           25.9.0
  shellcheck:      v0.11.0

════════════════════════════════════════════════════════
ℹ️  Go Installation
════════════════════════════════════════════════════════
ℹ️  Current Go version: 1.25.4
✅ Go 1.25 already installed

════════════════════════════════════════════════════════
ℹ️  Docker Check
════════════════════════════════════════════════════════
✅ Docker is installed and running: 29.0.2

════════════════════════════════════════════════════════
ℹ️  Python & Poetry Installation
════════════════════════════════════════════════════════
✅ Python installed: 3.12
✅ Poetry installed: 1.8.2
✅ Black installed: 25.9.0
3.12.3

════════════════════════════════════════════════════════
ℹ️  Development Tools Installation
════════════════════════════════════════════════════════
✅ Helm already installed: v3.19.2+g8766e71
✅ kubectl already installed: Client Version: v1.34.1
Kustomize Version: v5.7.1
ℹ️  Installing Protocol Buffers v33.0...
Continue? [Y/n] 
make: *** [Makefile:123: dev-env-setup] Error 8
```

--- a/DEVELOPMENT.md
+++ b/DEVELOPMENT.md
@@ -77,6 +77,19 @@ This will:
 make dev-env-setup AUTO_MODE=true
 ```
 
+**Debugging setup issues**: If the setup script fails, enable debug mode for detailed output:
+```bash
+DEBUG=true make dev-env-setup AUTO_MODE=true
+```
+
+This will show:
+- Architecture detection and mappings
+- URL construction for all downloads
+- HTTP response codes for failed downloads
+- Detailed error messages with suggestions
+
+Common setup issues and solutions are documented in the [Debugging](#-debugging) section.
+
 ### Build System
 
 **Unified build system** features:
@@ -850,6 +863,187 @@ The CI environment uses:
    # Ubuntu: apt-get install shellcheck
    # See: https://github.com/koalaman/shellcheck#installing
    ```
+
+### Debugging Setup Script Issues
+
+The `scripts/setup-dev-env.sh` script installs all development dependencies. If you encounter issues:
+
+#### Enable Debug Mode
+
+```bash
+# Run with detailed debugging output
+DEBUG=true make dev-env-setup AUTO_MODE=true
+
+# Or run the script directly
+DEBUG=true AUTO_MODE=true ./scripts/setup-dev-env.sh
+```
+
+**Debug output includes**:
+- Architecture detection (x86_64 → amd64, aarch64 → arm64)
+- Architecture mappings for different tools (GO_ARCH vs PROTOC_ARCH)
+- Complete URLs being constructed for downloads
+- HTTP response codes from URL verification
+- Version information from `.versions.yaml`
+
+#### Common Setup Issues
+
+**1. Download Failures (404 errors)**
+
+If you see errors like "HTTP 404 Not Found":
+
+```bash
+# Enable debug mode to see the exact URL
+DEBUG=true ./scripts/setup-dev-env.sh
+
+# Verify the URL manually
+curl -I "https://github.com/koalaman/shellcheck/releases/download/v0.11.0/shellcheck-v0.11.0.linux.x86_64.tar.xz"
+
+# Check if the release exists on GitHub
+# Visit: https://github.com/<owner>/<repo>/releases
+```
+
+**Common causes**:
+- Version in `.versions.yaml` doesn't exist in GitHub releases
+- Architecture-specific filename doesn't match release assets
+- Tool project changed their release naming convention
+
+**2. Architecture Mismatch**
+
+Different tools use different architecture naming:
+- **Go tools** (yq, kubectl, helm): `amd64`, `arm64`, `darwin`
+- **Protocol Buffers**: `x86_64`, `aarch_64`, `osx-universal_binary`
+- **Shellcheck**: `x86_64`, `aarch64`, `darwin`
+
+The script automatically maps these:
+```bash
+# See mappings in debug output
+DEBUG=true ./scripts/setup-dev-env.sh
+# Look for: "DEBUG: Architecture mappings: Raw ARCH: x86_64, GO_ARCH: amd64, PROTOC_ARCH: x86_64"
+```
+
+**3. Permission Issues**
+
+```bash
+# If installation to /usr/local/bin fails
+sudo make dev-env-setup AUTO_MODE=true
+
+# Or install to user directory (requires PATH modification)
+mkdir -p ~/bin
+export PATH="$HOME/bin:$PATH"
+# Modify script to use ~/bin instead of /usr/local/bin
+```
+
+**4. Network/Proxy Issues**
+
+```bash
+# Test connectivity to GitHub
+curl -I https://github.com
+
+# If behind proxy, configure:
+export HTTP_PROXY="http://proxy.example.com:8080"
+export HTTPS_PROXY="http://proxy.example.com:8080"
+export NO_PROXY="localhost,127.0.0.1"
+
+# Then retry
+DEBUG=true make dev-env-setup AUTO_MODE=true
+```
+
+**5. Version Validation**
+
+```bash
+# Check what versions are configured
+cat .versions.yaml
+
+# Verify specific tool version exists
+TOOL_VERSION=$(yq eval '.SHELLCHECK_VERSION' .versions.yaml)
+echo "Checking shellcheck version: $TOOL_VERSION"
+curl -I "https://github.com/koalaman/shellcheck/releases/tag/$TOOL_VERSION"
+
+# Update version if needed
+yq eval '.SHELLCHECK_VERSION = "v0.11.0"' -i .versions.yaml
+```
+
+#### Testing URL Construction
+
+To test URL construction without running the full setup:
+
+```bash
+# Source the script functions
+source scripts/setup-dev-env.sh
+
+# Test specific tool URLs
+echo "yq URL: $YQ_URL"
+echo "kubectl URL: $KUBECTL_URL"
+echo "protoc URL: $PROTOC_URL"
+echo "shellcheck URL: $SHELLCHECK_URL"
+
+# Test URL accessibility
+curl -I "$SHELLCHECK_URL"
+```
+
+#### Manual Verification Steps
+
+1. **Verify tool exists in releases**:
+   ```bash
+   # Check GitHub releases page
+   open "https://github.com/koalaman/shellcheck/releases"
+   
+   # Or use API
+   curl -s "https://api.github.com/repos/koalaman/shellcheck/releases/latest" | grep browser_download_url
+   ```
+
+2. **Test download manually**:
+   ```bash
+   # Download specific asset
+   wget "https://github.com/koalaman/shellcheck/releases/download/v0.11.0/shellcheck-v0.11.0.linux.x86_64.tar.xz"
+   
+   # Verify archive integrity
+   tar -tzf shellcheck-v0.11.0.linux.x86_64.tar.xz
+   ```
+
+3. **Validate script syntax**:
+   ```bash
+   # Check for syntax errors
+   bash -n scripts/setup-dev-env.sh
+   
+   # Run shellcheck on the script itself
+   shellcheck scripts/setup-dev-env.sh
+   ```
+
+#### Reporting Setup Issues
+
+When reporting setup script issues, include:
+
+1. **Debug output**:
+   ```bash
+   DEBUG=true make dev-env-setup AUTO_MODE=true 2>&1 | tee setup-debug.log
+   ```
+
+2. **Environment details**:
+   ```bash
+   echo "OS: $(uname -s)"
+   echo "Architecture: $(uname -m)"
+   echo "Shell: $SHELL"
+   echo "Bash version: $BASH_VERSION"
+   ```
+
+3. **Version file content**:
+   ```bash
+   cat .versions.yaml
+   ```
+
+4. **Failed URL** (from debug output):
+   ```
+   Look for lines like:
+   ❌ ERROR: Failed to verify URL: https://...
+   HTTP Status: 404 Not Found
+   ```
+
+This information helps diagnose whether the issue is:
+- Version-specific (tool version doesn't exist)
+- Architecture-specific (wrong filename for platform)
+- Network-related (connectivity or proxy issues)
+- Script bug (incorrect URL construction logic)
 
 ## 🔧 Makefile Reference
 

--- a/scripts/setup-dev-env.sh
+++ b/scripts/setup-dev-env.sh
@@ -39,7 +39,7 @@ BLUE='\033[0;34m'
 NC='\033[0m' # No Color
 
 # Configuration
-AUTO_MODE=false
+AUTO_MODE=${AUTO_MODE:-false}  # Preserve environment variable if set
 SKIP_GO=false
 SKIP_DOCKER=false
 SKIP_PYTHON=false
@@ -49,9 +49,17 @@ SKIP_TOOLS=false
 OS=$(uname -s | tr '[:upper:]' '[:lower:]')
 ARCH=$(uname -m)
 
+# Map architecture names for different use cases
 case "${ARCH}" in
-    x86_64) ARCH="amd64" ;;
-    aarch64|arm64) ARCH="arm64" ;;
+    x86_64) 
+        GO_ARCH="amd64"      # Go uses amd64
+        PROTOC_ARCH="x86_64" # Protocol Buffers uses x86_64
+        ;;
+    aarch64|arm64) 
+        GO_ARCH="arm64"
+        PROTOC_ARCH="aarch_64"  # Protocol Buffers uses aarch_64
+        ARCH="arm64"  # Normalize to arm64
+        ;;
     *) echo -e "${RED}❌ Unsupported architecture: ${ARCH}${NC}" && exit 1 ;;
 esac
 
@@ -70,6 +78,50 @@ log_warning() {
 
 log_error() {
     echo -e "${RED}❌ $*${NC}"
+}
+
+log_debug() {
+    if [[ "${DEBUG:-false}" == "true" ]]; then
+        echo -e "${BLUE}🔍 DEBUG: $*${NC}"
+    fi
+}
+
+# Function to verify and log download URLs
+verify_download_url() {
+    local url="$1"
+    local description="$2"
+    
+    log_debug "Attempting to verify: $description"
+    log_debug "URL: $url"
+    
+    # Test URL accessibility (without downloading the full file)
+    local http_code
+    http_code=$(curl -f -s -I -L -w "%{http_code}" -o /dev/null "$url" 2>/dev/null)
+    local curl_exit_code=$?
+    
+    if [[ $curl_exit_code -ne 0 || "$http_code" != "200" ]]; then
+        log_error "Failed to access URL for $description"
+        log_error "URL: $url"
+        log_error "HTTP status code: ${http_code:-unknown}"
+        log_error "curl exit code: $curl_exit_code"
+        
+        # Provide helpful suggestions based on the error
+        if [[ "$http_code" == "404" ]]; then
+            log_error "File not found (404). The version or architecture might not be available."
+            log_error "Suggestion: Check if the version exists in the GitHub releases page"
+        elif [[ "$http_code" == "403" ]]; then
+            log_error "Access forbidden (403). This might be a rate limit or authentication issue."
+        elif [[ $curl_exit_code -eq 6 ]]; then
+            log_error "Could not resolve host. Check your internet connection."
+        elif [[ $curl_exit_code -eq 7 ]]; then
+            log_error "Failed to connect to host. Check your internet connection and firewall."
+        fi
+        
+        return 1
+    fi
+    
+    log_debug "URL verified successfully for $description (HTTP $http_code)"
+    return 0
 }
 
 command_exists() {
@@ -105,6 +157,10 @@ OPTIONS:
     --skip-tools        Skip development tools installation
     --help              Show this help message
 
+ENVIRONMENT VARIABLES:
+    AUTO_MODE=true      Same as --auto flag (non-interactive mode)
+    DEBUG=true          Enable debug output (shows URLs, architecture mappings, etc.)
+
 EXAMPLES:
     # Interactive mode
     $0
@@ -114,6 +170,12 @@ EXAMPLES:
 
     # Install only tools (Go already installed)
     $0 --skip-go --skip-docker
+
+    # Enable debug output to troubleshoot download issues
+    DEBUG=true $0
+
+    # Automated setup with debug output
+    AUTO_MODE=true DEBUG=true $0
 
 EOF
 }
@@ -156,10 +218,14 @@ done
 # Banner
 echo ""
 echo "╔════════════════════════════════════════════════════════╗"
-echo "║   NVSentinel Development Environment Setup            ║"
+echo "║   NVSentinel Development Environment Setup             ║"
 echo "╚════════════════════════════════════════════════════════╝"
 echo ""
-log_info "Platform: ${OS}-${ARCH}"
+log_info "Platform: ${OS}-${GO_ARCH}"
+log_debug "Architecture mappings:"
+log_debug "  Raw ARCH: ${ARCH}"
+log_debug "  GO_ARCH (for Go tools, yq, kubectl): ${GO_ARCH}"
+log_debug "  PROTOC_ARCH (for Protocol Buffers): ${PROTOC_ARCH}"
 echo ""
 
 # Check if .versions.yaml exists
@@ -181,8 +247,18 @@ if ! command_exists yq; then
             exit 1
         fi
     elif [[ "${OS}" == "linux" ]]; then
-        sudo wget -qO /usr/local/bin/yq https://github.com/mikefarah/yq/releases/latest/download/yq_linux_"${ARCH}"
-        sudo chmod +x /usr/local/bin/yq
+        YQ_URL="https://github.com/mikefarah/yq/releases/latest/download/yq_linux_${GO_ARCH}"
+        log_debug "Architecture mapping: ${ARCH} -> GO_ARCH=${GO_ARCH}"
+        
+        if verify_download_url "$YQ_URL" "yq for Linux ${GO_ARCH}"; then
+            log_debug "Downloading yq from: $YQ_URL"
+            sudo wget -qO /usr/local/bin/yq "$YQ_URL"
+            sudo chmod +x /usr/local/bin/yq
+            log_debug "yq installed successfully to /usr/local/bin/yq"
+        else
+            log_error "Failed to install yq. Please install manually or check your architecture."
+            exit 1
+        fi
     fi
     
     log_success "yq installed"
@@ -364,9 +440,22 @@ if [[ "${SKIP_TOOLS}" == "false" ]]; then
             if [[ "${OS}" == "darwin" ]]; then
                 brew install kubectl
             elif [[ "${OS}" == "linux" ]]; then
+                log_debug "Fetching latest kubectl version..."
                 KUBECTL_VERSION=$(curl -L -s https://dl.k8s.io/release/stable.txt)
-                sudo curl -L "https://dl.k8s.io/release/${KUBECTL_VERSION}/bin/linux/${ARCH}/kubectl" -o /usr/local/bin/kubectl
-                sudo chmod +x /usr/local/bin/kubectl
+                KUBECTL_URL="https://dl.k8s.io/release/${KUBECTL_VERSION}/bin/linux/${GO_ARCH}/kubectl"
+                
+                log_debug "kubectl version: ${KUBECTL_VERSION}"
+                log_debug "Architecture mapping: ${ARCH} -> GO_ARCH=${GO_ARCH}"
+                
+                if verify_download_url "$KUBECTL_URL" "kubectl ${KUBECTL_VERSION} for Linux ${GO_ARCH}"; then
+                    log_debug "Downloading kubectl from: $KUBECTL_URL"
+                    sudo curl -L "$KUBECTL_URL" -o /usr/local/bin/kubectl
+                    sudo chmod +x /usr/local/bin/kubectl
+                    log_debug "kubectl installed successfully to /usr/local/bin/kubectl"
+                else
+                    log_error "Failed to install kubectl. Please install manually or check your architecture."
+                    exit 1
+                fi
             fi
             log_success "kubectl installed"
         fi
@@ -383,12 +472,24 @@ if [[ "${SKIP_TOOLS}" == "false" ]]; then
             if [[ "${OS}" == "darwin" ]]; then
                 PROTOC_ZIP="protoc-${PROTOBUF_VERSION_NUM}-osx-universal_binary.zip"
             elif [[ "${OS}" == "linux" ]]; then
-                PROTOC_ZIP="protoc-${PROTOBUF_VERSION_NUM}-linux-${ARCH}.zip"
+                PROTOC_ZIP="protoc-${PROTOBUF_VERSION_NUM}-linux-${PROTOC_ARCH}.zip"
             fi
             
-            TMP_DIR=$(mktemp -d)
-            cd "${TMP_DIR}"
-            wget -q "https://github.com/protocolbuffers/protobuf/releases/download/${PROTOBUF_VERSION}/${PROTOC_ZIP}"
+            PROTOC_URL="https://github.com/protocolbuffers/protobuf/releases/download/${PROTOBUF_VERSION}/${PROTOC_ZIP}"
+            
+            log_debug "protobuf version: ${PROTOBUF_VERSION} (${PROTOBUF_VERSION_NUM})"
+            log_debug "Architecture mapping: ${ARCH} -> PROTOC_ARCH=${PROTOC_ARCH}"
+            log_debug "Expected file: ${PROTOC_ZIP}"
+            
+            if verify_download_url "$PROTOC_URL" "protoc ${PROTOBUF_VERSION} for ${OS} ${PROTOC_ARCH}"; then
+                TMP_DIR=$(mktemp -d)
+                cd "${TMP_DIR}"
+                log_debug "Downloading protoc from: $PROTOC_URL"
+                wget -q "$PROTOC_URL"
+            else
+                log_error "Failed to download protoc. Please check the version and architecture."
+                exit 1
+            fi
             unzip -q "${PROTOC_ZIP}"
             sudo cp bin/protoc /usr/local/bin/
             sudo mkdir -p /usr/local/include
@@ -406,16 +507,33 @@ if [[ "${SKIP_TOOLS}" == "false" ]]; then
     else
         log_info "Installing shellcheck ${SHELLCHECK_VERSION}..."
         if prompt_continue; then
-            SHELLCHECK_VERSION_NUM=${SHELLCHECK_VERSION#v}
-            
             if [[ "${OS}" == "darwin" ]]; then
                 brew install shellcheck
             elif [[ "${OS}" == "linux" ]]; then
                 TMP_DIR=$(mktemp -d)
                 cd "${TMP_DIR}"
-                wget -q "https://github.com/koalaman/shellcheck/releases/download/${SHELLCHECK_VERSION}/shellcheck-${SHELLCHECK_VERSION_NUM}.linux.${ARCH}.tar.xz"
-                tar -xJ -f "shellcheck-${SHELLCHECK_VERSION_NUM}.linux.${ARCH}.tar.xz"
-                sudo cp "shellcheck-${SHELLCHECK_VERSION_NUM}/shellcheck" /usr/local/bin/
+                # Shellcheck uses x86_64 for amd64 and aarch64 for arm64
+                SHELLCHECK_ARCH=${PROTOC_ARCH}
+                if [[ "${ARCH}" == "arm64" ]]; then
+                    SHELLCHECK_ARCH="aarch64"
+                fi
+                
+                SHELLCHECK_FILE="shellcheck-${SHELLCHECK_VERSION}.linux.${SHELLCHECK_ARCH}.tar.xz"
+                SHELLCHECK_URL="https://github.com/koalaman/shellcheck/releases/download/${SHELLCHECK_VERSION}/${SHELLCHECK_FILE}"
+                
+                log_debug "shellcheck version: ${SHELLCHECK_VERSION}"
+                log_debug "Architecture mapping: ${ARCH} -> PROTOC_ARCH=${PROTOC_ARCH} -> SHELLCHECK_ARCH=${SHELLCHECK_ARCH}"
+                log_debug "Expected file: ${SHELLCHECK_FILE}"
+                
+                if verify_download_url "$SHELLCHECK_URL" "shellcheck ${SHELLCHECK_VERSION} for Linux ${SHELLCHECK_ARCH}"; then
+                    log_debug "Downloading shellcheck from: $SHELLCHECK_URL"
+                    wget -q "$SHELLCHECK_URL"
+                else
+                    log_error "Failed to download shellcheck. Please check the version and architecture."
+                    exit 1
+                fi
+                tar -xJ -f "shellcheck-${SHELLCHECK_VERSION}.linux.${SHELLCHECK_ARCH}.tar.xz"
+                sudo cp "shellcheck-${SHELLCHECK_VERSION}/shellcheck" /usr/local/bin/
                 sudo chmod +x /usr/local/bin/shellcheck
                 cd - >/dev/null
                 rm -rf "${TMP_DIR}"
@@ -514,7 +632,7 @@ if [[ "${SKIP_PYTHON}" == "false" ]] && command_exists python3; then
     log_info "Python gRPC Tools"
     echo "════════════════════════════════════════════════════════"
     
-    log_info "Installing Python gRPC tools (grpcio, grpcio-tools)..."
+    log_info "Installing Python gRPC tools: grpcio, grpcio-tools..."
     
     if prompt_continue; then
         if [[ "${OS}" == "darwin" ]]; then

